### PR TITLE
issue fixed

### PR DESCRIPTION
**Pull request by qazi-fahid**
### Description
Profile image gets broken whenever we edit the username

The issue is fixed by renaming the uploaded image files for all four sizes according to the new username.

### Testing
- [ ] Run coverage command and mention new % coverage
- [ ] This PR ensures security is maintained w.r.t input validations, output escaping/encoding and privileges against user roles

### Performance
- [ ] Can impact performance (Flag if these changes can affect app performance and add a corresponding JIRA ticket here to conduct performance testing)

### Dependencies
- [ ] This PR is dependent on any other PR or configuration or JIRA issue number; please mention here
- [ ] Includes migration
 
### Post-review
- [ ] Rebase and squash commits before merge

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @qazi-fahid! I've created [OSPR-3231](https://openedx.atlassian.net/browse/OSPR-3231) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://open.edx.org/wp-content/uploads/2019/01/individual-contributor-agreement.pdf)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

File: edx_solutions_api_integration/users/views.py [hunk 1]
```diff
@@ -732,6 +734,21 @@ def post(self, request, user_id):  # pylint: disable=R0915
         existing_user.save()
 
         username = request.data.get('username', None)
+        old_username = str(existing_user.username)
```
Comment (msaqib52) [inline edx_solutions_api_integration/users/views.py]: @qazi-fahid this block should run only when username gets updated, please see [L765](https://github.com/edx-solutions/api-integration/pull/214/files#diff-6e10f5856db69f12fe0c2aef7e4ec7dfR765)

Comment (qazi-fahid) [reply edx_solutions_api_integration/users/views.py]: done now the code runs after the username gets updated and also done the version bump

File: edx_solutions_api_integration/users/views.py [hunk 2]
```diff
@@ -92,6 +93,7 @@
     UserRolesSerializer,
     CourseProgressSerializer,
 )
+from openedx.core.djangoapps.user_api.accounts.image_helpers import get_profile_image_names, get_profile_image_storage
```
Comment (naeem91) [inline edx_solutions_api_integration/users/views.py]: can you place it along with other openedx imports?

File: edx_solutions_api_integration/users/views.py [hunk 3]
```diff
@@ -749,6 +752,21 @@ def post(self, request, user_id):  # pylint: disable=R0915
             response_data['username'] = existing_user.username
             existing_user.save()
 
+            if old_username != username:
+                storage = get_profile_image_storage()
+                profile_image_names = get_profile_image_names(old_username)
+                new_profile_image_names = get_profile_image_names(username)
+                for old_image_size, old_image_name in profile_image_names.items():
```
Comment (naeem91) [inline edx_solutions_api_integration/users/views.py]: Nested looping can be avoided here: 
```python
for old_image_size, old_image_name in profile_image_names.items():
  new_image_name = new_profile_image_names.get(old_image_size)
  if new_image_name:
     new_image_path = os.path.join(storage.location, new_image_name)
   ....
```

File: edx_solutions_api_integration/users/views.py (hunk 2 above)
Comment (qazi-fahid) [reply edx_solutions_api_integration/users/views.py]: Done

File: edx_solutions_api_integration/users/views.py (hunk 3 above)
Comment (qazi-fahid) [reply edx_solutions_api_integration/users/views.py]: Done

File: edx_solutions_api_integration/users/views.py [hunk 4]
```diff
@@ -750,6 +753,19 @@ def post(self, request, user_id):  # pylint: disable=R0915
...
+                    if storage.exists(old_image_name):
+                        old_image_path = storage.path(old_image_name)
+                        new_image_path = storage.location + '/' + new_profile_image_names[old_image_size]
```
Comment (naeem91) [inline edx_solutions_api_integration/users/views.py]: Please use os.path.join to concatenate paths. 
```
new_image_path = os.path.join(storage.location, new_image_name)
```

Comment (qazi-fahid) [reply edx_solutions_api_integration/users/views.py]: done

**Pull request merged**